### ISSUE TITLE
save model-----best/last

### ISSUE THREAD

**Issue by zjp99** [xingyizhou/CenterNet#468]
model_best is the model with the lowest loss. model_last is the best in mAP ??

**Comment by zjp99** [comment]
voc07+12，but the results of AP is different.best model  map==0.7331,last model:Mean AP = 0.7416,I use the model you provided,ap=== 0.7764,add  --flip_test map==0.7932.
How can I achieve this result?which model should Iuse?(my training model)

**Comment by xingyizhou** [comment]
You are correct. model_last is the model in the last epoch. Make sure you have trained the full schedule.

**Comment by zjp99** [comment]
> You are correct. model_last is the model in the last epoch. Make sure you have trained the full schedule.

Thank you!
python main.py ctdet --exp_id pascal_dla_384 --dataset pascal  --num_epochs 70 --lr_step 45,60 --resume --flip_test
I use this command to train. Test on the last model (70 epoch).but  Map==74.16.If normal training can get the best results map==0.7932?Do I need to adjust something or retrain?

**Comment by zjp99** [comment]
my batch_size is 8.Does it matter?

**Comment by xingyizhou** [comment]
It matters a lot.

**Comment by laclouis5** [comment]
Hi everyone,

@xingyizhou it appears that model_best is the best according to 'loss' metric. I think it  would be more relevant to retain the model with the best mAP@.5 / Coco AP, why choosing loss?

**Comment by zjp99** [comment]
Hello, I want to ask if your code is the code in the paper.I train VOC.object detection batch_size:8 map：74.16.and batch_size：16 map:74.69.If my batch_size is 32, will the result reach 79.3？？
       I want to ask if the code is final? Or do I need to change anything?

**Comment by wilsoncwj** [comment]
@xingyizhou Is there a way to find out which epoch gave the best model?

**Comment by vinthony** [comment]
Hi everyone, are there any new comments on this problem? which model should be used when testing?

**Comment by sisrfeng** [comment]
> Hi everyone,
> 
> @xingyizhou it appears that model_best is the best according to 'loss' metric. I think it would be more relevant to retain the model with the best mAP@.5 / Coco AP, why choosing loss?

That' s what I use now~

**Comment by sisrfeng** [comment]
> It matters a lot.

How to choose the optimal batch_size?

**Comment by sisrfeng** [comment]
model_best will be  replaced by a model even if it performs badly when we resume an experiment, because once we restart the main.py,  
'best = 1e10'
will take effect again.
calculate the 
`log_dict_val[opt.metric]`
and set
'if log_dict_val[opt.metric]<best:
      best = log_dict_val[opt.metric]'
right after loading a model can solve the problem, but waste time
